### PR TITLE
Implement create, put resource methods

### PR DESCRIPTION
**Pull request by AnjanaSenanayake**
Implement create, put resource methods

### CODE REVIEW

File: choreo-apis/practitioner/service.bal [hunk 1]
```diff
@@ -131,13 +131,83 @@ service / on new fhirr4:Listener(9090, apiConfig) {
...
+            anydata parsedPayload = check fhirParser:parse(practitioner.toJson());
+            lkcore010:LKCorePractitioner parsedResource = check parsedPayload.cloneWithType(lkcore010:LKCorePractitioner);
+
+            http:Response|http:ClientError response = practitionerApiClient->put(path, message = parsedResource, 
+                headers = {
+                    apikey: apiKey
+                }
+            );
+
+            if (response is http:Response) {
```
Comment (sameeragunarathne) [inline choreo-apis/practitioner/service.bal]: Shall we do early returns

File: choreo-apis/practitioner/service.bal [hunk 2]
```diff
@@ -131,13 +131,83 @@ service / on new fhirr4:Listener(9090, apiConfig) {
...
+                            return check result.cloneWithType(lkcore010:LKCorePractitioner);
+                        } on fail var e {
+                            return handleError("Error occurred while parsing the response: ",e, http:STATUS_UNPROCESSABLE_ENTITY);
+                        }
+                    } else {
+                        return handleError("Error occurred while accessing response: ",jsonPayload, http:STATUS_UNPROCESSABLE_ENTITY);
+                    }
+                }
+                return handleError("Error code returned from the API: ",jsonPayload, http:STATUS_INTERNAL_SERVER_ERROR);
+            }
```
Comment (sameeragunarathne) [inline choreo-apis/practitioner/service.bal]: Let's see if we can common out this repeating logic to a util. Let's pass the interaction information to the error messages as well.

File: choreo-apis/patient/util.bal [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import ballerinax/health.fhir.r4;
+import ballerina/log;
+isolated function isSuccessCode(int statusCode) returns boolean {
```
Comment (sameeragunarathne) [inline choreo-apis/patient/util.bal:4]: Let's properly format

File: choreo-apis/practitioner/service.bal (hunk 1 above)
Comment (AnjanaSenanayake) [reply choreo-apis/practitioner/service.bal]: Fixed in https://github.com/lk-gov-health-hiu/project-nhdx/commit/928ffb84a2edd972a5ceeb9380b0560753ec29b3

File: choreo-apis/patient/util.bal (hunk 3 above)
Comment (AnjanaSenanayake) [reply choreo-apis/patient/util.bal:4]: Fixed in https://github.com/lk-gov-health-hiu/project-nhdx/commit/928ffb84a2edd972a5ceeb9380b0560753ec29b3

File: choreo-apis/practitioner/service.bal (hunk 2 above)
Comment (AnjanaSenanayake) [reply choreo-apis/practitioner/service.bal]: Kept as it is, as discussed offline.

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**